pkg/controller/v1beta1/inferenceservice/reconcilers/deployment/deployment_reconciler.go

```diff
@@ -0,0 +1,191 @@
+/*
+Copyright 2021 kubeflow.org.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deployment
+
+import (
+	"context"
+
+	"github.com/kubeflow/kfserving/pkg/apis/serving/v1beta1"
+	"github.com/kubeflow/kfserving/pkg/constants"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var log = logf.Log.WithName("DeploymentReconciler")
+
+//DeploymentReconciler is the struct of Raw K8S Object
+type DeploymentReconciler struct {
+	client          client.Client
+	scheme          *runtime.Scheme
+	Deployment      *appsv1.Deployment
+	componentExt    *v1beta1.ComponentExtensionSpec
+	componentStatus v1beta1.ComponentStatusSpec
+}
+
+func NewDeploymentReconciler(client client.Client,
+	scheme *runtime.Scheme,
+	componentMeta metav1.ObjectMeta,
+	componentExt *v1beta1.ComponentExtensionSpec,
+	podSpec *corev1.PodSpec,
+	componentStatus v1beta1.ComponentStatusSpec) *DeploymentReconciler {
+	return &DeploymentReconciler{
+		client:          client,
+		scheme:          scheme,
+		Deployment:      createRawDeployment(componentMeta, componentExt, podSpec, componentStatus),
+		componentExt:    componentExt,
+		componentStatus: componentStatus,
+	}
+}
+
+func createRawDeployment(componentMeta metav1.ObjectMeta,
+	componentExt *v1beta1.ComponentExtensionSpec,
+	podSpec *corev1.PodSpec,
+	componentStatus v1beta1.ComponentStatusSpec) *appsv1.Deployment {
+	var minReplicas int32
+	if componentExt.MinReplicas == nil {
+		minReplicas = int32(constants.DefaultMinReplicas)
+	} else {
+		minReplicas = int32(*componentExt.MinReplicas)
+	}
+
+	if minReplicas < int32(constants.DefaultMinReplicas) {
+		minReplicas = int32(constants.DefaultMinReplicas)
+	}
+	setDefaultPodSpec(podSpec)
+	deployment := &appsv1.Deployment{
+		ObjectMeta: componentMeta,
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &minReplicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": constants.GetRawServiceLabel(componentMeta.Name),
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": constants.GetRawServiceLabel(componentMeta.Name),
+					},
```

Comment:
We need to pass the labels and annotations from `componentMeta` to the template, the pod mutator is checking [the annotations](https://github.com/kubeflow/kfserving/blob/master/pkg/webhook/admission/pod/storage_initializer_injector.go#L81) to inject the storage initializer.

Reply:
got it, thanks